base_user_role/migrations/12.0.3.0.0/post-migration.py

```diff
@@ -0,0 +1,21 @@
+# Copyright (C) 2021 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+from odoo import api, SUPERUSER_ID
+
+_logger = logging.getLogger(__name__)
+
+
+def _affect_group_to_category(env):
+    _logger.info("Set Category 'User roles' to groups related to roles")
+    roles = env["res.users.role"].search([])
```

Comment:
Here too, only select roles that have no category set please.

Reply:
Hi @StefanRijnhart it is handled in the line 13. Or I missed something ? ``.filtered(lambda x: not x.category_id)``

Reply:
Indeed! Sorry, did not look carefully.